src/promotion/nhit/nhit_hash.c

```diff
@@ -163,7 +163,10 @@ ocf_error_t nhit_hash_init(uint64_t hash_size, nhit_hash_t *ctx)
 		env_atomic_set(&new_ctx->ring_buffer[i].counter, 0);
 	}
 
-	env_spinlock_init(&new_ctx->rb_pointer_lock);
+	result = env_spinlock_init(&new_ctx->rb_pointer_lock);
+	if (result)
+		goto dealloc_locks;
```

Comment:
This won't work. i_locks is set to new_ctx->hash_entries, there will be out of bound access to array at line 177